src/Paginator.php

```diff
@@ -12,7 +12,7 @@
 use Laminas\Db\ResultSet\AbstractResultSet;
 use Laminas\Filter\FilterInterface;
 use Laminas\Paginator\Adapter\AdapterInterface;
-use Laminas\Paginator\Adapter\DbSelect;
+use Laminas\Paginator\Adapter\ArrayCopyInterface;
```

Comment:
This is a **new** interface that you declared: this means that nobody implements it.


As mentioned during discussions, a test would highlight the issue you are having :+1:

Reply:
Note that the idea behind the interface is to prevent the need of laminas-paginator-adapter-laminasdb as a dependency.

Reply:
Yes, but that doesn't work for existing `DbSelect` instances passed in :-) 

Nobody has an `ArrayCopyInterface` implementation, so the introduced code-path is "dead" for all consumers, at the time we would introduce this change in a release.